### ISSUE TITLE
Implement Adaptive Contextual Masking Training (CMT) with Dynamic Intensity

### ISSUE THREAD

**Issue by C12ux** [haze103/web-dev-laravel-proj#35]
## Feature: Adaptive Dynamic Masking for CMT

**Background:**
Current Contextual Masking Training (CMT) uses fixed parameters for masking probability and intensity based on the training phase or epoch. To make CMT more robust and efficient, introduce an adaptive masking that dynamically adjusts its intensity (masking probability, number/size of masks, etc.) based on validation loss during training.

### Requirements
- Monitor validation loss after each epoch.
- Increase masking intensity when validation loss plateaus or decreases slowly (model getting comfortable).
- Decrease masking intensity if validation loss spikes (model struggling to generalize).
- Masking intensity parameters to adjust:
    - `mask_prob` (masking probability per image)
    - `mask_size_range` and/or number of masks
    - Optionally, mask type or fill_value
- Integrate seamlessly into existing training loop, allowing the policy to update intensity parameters per epoch based on validation metrics.
- Document the adaptive logic in code and README.

### Example logic (Python pseudocode):
```python
# After each validation epoch:
if val_loss < prev_val_loss - epsilon:
    # Loss improving: slowly increase masking
delta_mask = min(max_step, rate * (prev_val_loss - val_loss))
adaptive_mask_prob += delta_mask
elif val_loss > prev_val_loss + epsilon:
    # Loss worsened: ease masking
adaptive_mask_prob = max(min_prob, adaptive_mask_prob - max_step)
# Clamp within [min_prob, max_prob]
```

- Optionally, use moving average of validation loss to smooth out noisy fluctuations.
- Save and log adaptive parameter history for reproducibility.
- Optionally: export plots showing how masking intensity evolved as loss changed.

### Acceptance Criteria
- Training loop dynamically adapts masking parameters based on validation loss.
- Code and README are updated to reflect adaptive strategy.
- Initial/maximum/minimum intensity boundaries are parameterized and documented.
- Adaptive curve can be visualized from logs.

---
*Labels: enhancement, documentation*
